### PR TITLE
Ciam 6318 reintroduce aud validation

### PR DESCRIPTION
**Pull request by DGuhr**
## Describe your changes

- re-introduce audience checking
- internal config is already set up with the right aud claim to check, it's just not used as we had problems getting the right audience
- adds a few tests so we make sure multiple audiences are used and parsed correctly, even if we switch the lib sometime in the future

## Ticket reference (if applicable)
Fixes #

## Checklist

* [x] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [x] Do your changes have passing automated tests and sufficient observability?

* [x] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [x] Are the agreed upon coding/architectural practices applied?

* [x] Are security needs fullfilled? (e.g. no internal URL)

* [x] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [-] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

**Comment by merlante** [comment]
LGTM -- just want to wait to see if we can fix the seats admin console issue before merging.

**Review by wscalf — APPROVED** [review]

**Pull request merged**
